apps/src/templates/sectionProgressV2/MoreDetailsDialog.jsx

```diff
@@ -1,27 +1,118 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import i18n from '@cdo/locale';
-import Button from '@cdo/apps/templates/Button';
 import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
-import {Heading3, Heading5} from '@cdo/apps/componentLibrary/typography';
+import {
+  Heading3,
+  Heading6,
+  StrongText,
+  BodyThreeText,
+} from '@cdo/apps/componentLibrary/typography';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import {ITEM_TYPE} from './ItemType';
+import ProgressIcon from './ProgressIcon';
+import styles from './progress-key-popup.module.scss';
 
 export default function MoreDetailsDialog({hasValidation, onClose}) {
+  const validatedLevelsContent = () => (
```

Comment:
Note: I thought about organizing all of the content into smaller functions like this so that the return would just list the functions rather than all of this html... I opted to not do that, but am interested in your $0.02 about readability and clean code overall.

I chose to put this into a function because it is rendered conditionally.

Reply:
What about a function to clean it up a bit:

```
const renderItem = (itemType, itemTitle, itemDetails) => 
      <div className={styles.item}>
          <ProgressIcon itemType={itemType} />
          (<BodyThreeText>
            <StrongText>{itemTitle + ': '}</StrongText>
            {itemDetails}
          </BodyThreeText>
        </div>);
 ```
 
Then each item would just need to be `renderItem(ITEM_TYPE.NOT_STARTED, i18n.notStarted(), i18n.progressLegendDetailsNotStarted()`

Reply:
I think that should probably clean up enough, but if you still want to make it a bit nicer, you could also split each header into a separate function. i.e. have `renderAssignmentCompletionStates()` and `renderTeacherActions()`

Reply:
Adding the `renderItem` function is perfect!  I am not quite as convinced that the `renderAssignmentCompletionStates()` is quite as needed... it would make it more readable but I don't know if it makes it really easier to refactor later on.